src/vs/workbench/contrib/terminal/common/terminalEnvironment.ts

```diff
@@ -444,7 +445,10 @@ export function injectShellIntegrationArgs(logService: ILogService, enableShellI
 	} else {
 		switch (shell) {
 			case 'bash':
-				if (!originalArgs || originalArgs.length === 0 || areZshBashLoginArgs(originalArgs)) {
+				if (!originalArgs || originalArgs.length === 0) {
+					newArgs = shellIntegrationArgs.get(ShellIntegrationExecutable.Bash);
+				} else if (areZshBashLoginArgs(originalArgs)) {
+					env['OS_IS_MAC'] = os === OperatingSystem.Macintosh ? '1' : '';
```

Comment:
We should keep the `VSCODE_SHELL_` prefix imo since the environment is very much a shared thing, sticking to LOGIN as well is probably a good idea as that's where we want to be eventually